### PR TITLE
feat(icon): removed all duplicate icon classes

### LINKED ISSUE #2339: ebay-icon: remove all duplicate classes
### Description

We should remove all classes from ebay-icon and use generic `icon-20` for width/height 20px. 
This will reduce the file size as well. 

### Screenshots

_No response_

### Figma link

_No response_

### Engineering pre-requisites

- [ ] Successful design review with Design System Team.
- [ ] Supports dark mode.
- [ ] Supports responsive design.
- [ ] Tokens have been provided (or already exist).
- [ ] Successful accessibility review with Core A11Y Team.
- [ ] Successful frontend review with eBayUI Team.

### PR DESCRIPTION
**Pull request by agliga**
Fixes #2339

- [X] This PR contains CSS changes
- [ ] This PR does not contain CSS changes

## Description
* Removed all icon--{name} classes
* Added icon--{size} classes
* Used generate SVG to do so
* For colored, used only height which should maintain aspect ratio

__note__ I will do visual regression once we get a thumbs up for this approach

## Checklist
- [X] I verify the build is in a non-broken state
- [X] I verify all changes are within scope of the linked issue

- [X] I regenerated all CSS files under dist folder
- [X] I tested the UI in all supported browsers
- [ ] I did a visual regression check of the components impacted by doing a Percy build and approved the build
- [X] I tested the UI in dark mode and RTL mode
- [X] I added/updated/removed Storybook coverage as appropriate

### CODE REVIEW

File: src/sass/icon/icon.scss
Comment (ArtBlue) [inline src/sass/icon/icon.scss:1]: The generated `sass` file (`src/sass/icon/generated/icon.scss`) and build (`dist/icon/icon.css`) still have classes with specific names, like this:

`svg.icon--chevron-right-12`

Shouldn't those now have the generic classes only?

**Review by ArtBlue — COMMENTED** [review]
Looks good overall, but some stuff...

File: src/sass/icon/icon.scss
Comment (agliga) [reply src/sass/icon/icon.scss:1]: That file is unused. I dont think we are importing it anywhere.

Comment (ArtBlue) [reply src/sass/icon/icon.scss:1]: Should these files not be deleted in that case?

Comment (agliga) [reply src/sass/icon/icon.scss:1]: Removed it

File: dist/icon/icon.css [hunk 1]
```diff
@@ -1,3208 +1,3 @@
...
-svg.star-rating--3,
-svg.star-rating--3-5,
-svg.star-rating--4,
-svg.star-rating--4-5,
-svg.star-rating--5 {
-    height: 24px;
-    width: 132px;
-}
-
-svg.star-rating--small-0,
```
Comment (ArtBlue) [inline dist/icon/icon.css:3192]: It looks like `star-rating-select` stopped working on Chrome. 

<kbd>image</kbd>

**Review by ArtBlue — CHANGES_REQUESTED** [review]
Looks good. I found one last issue.

File: dist/icon/icon.css (hunk 1 above)
Comment (agliga) [reply dist/icon/icon.css:3192]: Fixed it! It was a sass issue

**Review by ArtBlue — APPROVED** [review]
Nice.

**Pull request merged**